### PR TITLE
[Breaking] PyPi publish changes - unleash the Poetry!

### PR DESCRIPTION
**Pull request by osfanbuff63**
Whew, here's a quick summary of all the breaking changes...

- Source directory is now `vanilla_installer` instead of `vanilla-installer` (Mentioned on Discord in an internal channel)
- Dependency management has been completely migrated to [`poetry`](https://python-poetry.org)
  - `envs/` no longer exists, as it's unneeded
  - It's much easier to manage more than 1 Python version...
  - ...as it utilizes the [`pyproject.toml`](https://pip.pypa.io/en/stable/reference/build-system/pyproject-toml), using [Poetry's expansions](https://python-poetry.org/docs/pyproject) which allows a rule such as `>=3.8,<=3.11`.
- To install the GUI, you must install it with the `gui` extra
- Dependencies for docs have been split into the `docs` group, and development dependencies in the `dev` group
  - These can be installed with `poetry install --with <group>`. [Source from Poetry's docs](https://python-poetry.org/docs/cli/#install)
- You can no longer run the program using `python cli.py` or `python gui.py`
  - This is due to requirements for packaging. When you run `poetry install`, it installs the current project as well, as long as you do not specify `--no-root`. This allows for the program to be run as `vanilla-installer` as long as you have the venv checked out.
- Probably more but I'm tired and don't have time to write a longer PR description 😅

This is majorly for #21, but using `poetry` instead of `pipenv` if we're going to publish is a convenience thing, as it uses `pyproject.toml`.

### CODE REVIEW

**Comment by Kichura** [comment]
The GitHub Actions script is giving alerts about sensitive data being included, You reckon to investigate that?

**Comment by osfanbuff63** [comment]
Looks like it's upstream, will investigate more tomorrow.

**Comment by osfanbuff63** [comment]
Confirmed it's an upstream issue with CodeQL not liking requests. It was only finding it in `venv` code though, so this may or may not be relevant. Surprising that it only shows up *now* though.

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "vanilla-installer"
-version = "v1.0.0-dev6"
+version = "v1.0.0-dev7"
 description = "An installer for Fabulously Optimized for the vanilla launcher."
```
Comment (Kichura) [inline pyproject.toml]: The tone doesn't seem to make sense to me, Shouldn't it be `An installer of Fabulously Optimized for the vanilla launcher.` ?

File: vanilla_installer/gui.py [hunk 2]
```diff
@@ -373,36 +408,43 @@ def setupUi(self) -> None:
...
         )
         self.userLabel.setText(
-            QCoreApplication.translate("Dialog", "Github Username:", None)
+            QCoreApplication.translate("Dialog", "GitHub Username:", None)
         )
         self.loginInfoLabel.setText(
             QCoreApplication.translate(
                 "Dialog",
-                "Set your GitHub token to a personal token in order to extend your API ratelimit\n If you don't know what that is and aren't going to install Fabulously Optimized more than 60 times in an hour, You probably don't need this",
+                "Set your GitHub token to a personal token in order to extend your API ratelimit.\n If you don't know what that is and aren't going to install Fabulously Optimized more than 60 times in an hour, You probably don't need this",
```
Comment (Kichura) [inline vanilla_installer/gui.py]: `ratelimit` should have their words seperated iirc.

**Review by Kichura — COMMENTED** [review]
Very good although some questions are yet to be asked.

File: pyproject.toml (hunk 1 above)
Comment (osfanbuff63) [reply pyproject.toml]: Yeah that seems better, I'm not too good at these lol

**Review by Kichura — APPROVED** [review]

**Comment by nsde** [comment]
Is this ready to be merged?

**Comment by osfanbuff63** [comment]
If you think so, I think this could be ready, after all saves some merge conflicts later lol

**Pull request merged**